pkg/version/version.go

```diff
@@ -99,44 +99,67 @@ func CheckClusterVersion(ctx context.Context, client pd.Client) error {
 		}
 
 		tikvVersionString := removeVAndHash(s.Version)
-		tikvVersion, err := semver.NewVersion(tikvVersionString)
-		if err != nil {
-			return errors.Annotatef(berrors.ErrVersionMismatch, "%s: TiKV node %s version %s is invalid", err, s.Address, tikvVersionString)
+		tikvVersion, getVersionErr := semver.NewVersion(tikvVersionString)
+		if getVersionErr != nil {
+			return errors.Annotatef(berrors.ErrVersionMismatch, "%s: TiKV node %s version %s is invalid", getVersionErr, s.Address, tikvVersionString)
 		}
-
-		if tikvVersion.Compare(*minTiKVVersion) < 0 {
-			return errors.Annotatef(berrors.ErrVersionMismatch, "TiKV node %s version %s don't support BR, please upgrade cluster to %s",
-				s.Address, tikvVersionString, build.ReleaseVersion)
+		if checkerErr := checker(s, tikvVersion); checkerErr != nil {
+			return checkerErr
 		}
+	}
+	return nil
+}
 
-		if tikvVersion.Major != BRVersion.Major {
-			return errors.Annotatef(berrors.ErrVersionMismatch, "TiKV node %s version %s and BR %s major version mismatch, please use the same version of BR",
-				s.Address, tikvVersionString, build.ReleaseVersion)
+// CheckVersionForBackup checks the version for backup and
+func CheckVersionForBackup(backupVersion *semver.Version) VerChecker {
+	return func(store *metapb.Store, ver *semver.Version) error {
+		if backupVersion.Major != ver.Major {
```

Comment:
How about change it to `backupVersion.Major < ver.Major`?

Reply:
sounds good